src/include/txn_inline.h

```diff
@@ -1095,7 +1099,7 @@ __wt_txn_autocommit_check(WT_SESSION_IMPL *session)
     txn = session->txn;
     if (F_ISSET(txn, WT_TXN_AUTOCOMMIT)) {
         F_CLR(txn, WT_TXN_AUTOCOMMIT);
-        return (__wt_txn_begin(session, NULL));
+        return (__wt_txn_begin(session, NULL, true));
```

Comment:
Could you avoid needing the new flag by moving the clear of `WT_TXN_AUTOCOMMIT` until after the `wt_txn_begin` call?
I took a cursory look through where `WT_TXN_AUTOCOMMIT` is relevant, and I *don't think* that would change the behavior (i.e: that's definitely worth double checking).

Reply:
Yes, not clearing the flag doesn't have any problem. Updated the code accordingly.